src/Symfony/Component/Messenger/Command/ConsumeMessagesCommand.php

```diff
@@ -271,7 +271,7 @@ public function handleSignal(int $signal, int|false $previousExitCode = 0): int|
 
         $this->worker->stop();
 
-        return 0;
+        return false;
```

Comment:
There's no need to exit here, we just have to call `$this->worker->stop()` and the command will finish gracefully once the handler is done.